test/services/keyVault/keyVault-secret-tests.js

```
@@ -0,0 +1,377 @@
+    suiteUtil.setupSuite(done);
+  });
+
+  after(function (done) {
+    suiteUtil.teardownSuite(done);
+  });
+
+  beforeEach(function (done) {
+    suiteUtil.setupTest(done);
+  });
+
+  afterEach(function (done) {
+    suiteUtil.baseTeardownTest(done);
+  });
+  
+  describe('identifier', function() {
+    it('should work', function(done) {
+      
+      function assertMatch(vault, name, version, secretId) {
+        assertExactly(util.format('%s/secrets/%s', vault, name), secretId.baseIdentifier);
+        if (version) {
+          assertExactly(util.format('%s/secrets/%s/%s', vault, name, version), secretId.identifier);
+        } else {
+          assertExactly(secretId.baseIdentifier, secretId.identifier);
+        }
+        assertExactly(vault, secretId.vault);
+        assertExactly(name, secretId.name);
+        assertExactly(version, secretId.version);
+      }
+
+      function verifyCreate(vault, name, version) {
+        var secretId, parsedId;
+        if (version) {
+          secretId = KeyVault.createSecretIdentifier(vault, name, version);
+        } else {
+          secretId = KeyVault.createSecretIdentifier(vault, name);
+        }
+        assertMatch(vault, name, version, secretId);
+        if (version) {
+          parsedId = KeyVault.parseSecretIdentifier(secretId.identifier);
+          assertMatch(vault, name, version, parsedId);
+        }
+        parsedId = KeyVault.parseSecretIdentifier(secretId.baseIdentifier);
+        assertMatch(vault, name, null, parsedId);
+      }
+
+      verifyCreate(vaultUri, SECRET_NAME, null);
+      verifyCreate(vaultUri, SECRET_NAME, '1234');
+      
+      done();
+      
+    });
+  });
+  
+  describe('CRUD operations', function () {
+    it('should work', function (done) {
+      
+      var createdBundle;
+      var secretId;
+
+      function createSecret(next) {
+        client.setSecret(vaultUri, SECRET_NAME, { value: SECRET_VALUE }, function(err, secretBundle) {
+          if (err) throw err;
```

I prefer throwing. This is the fail fast approach. The stacktrace becomes more helpful.